services/QuillLMS/app/controllers/concerns/teacher_fixes.rb

```diff
@@ -48,6 +48,11 @@ def self.merge_activity_sessions_between_two_classroom_units(cu1, cu2)
       act_sesh2 = ActivitySession.find_by(classroom_unit: cu2, activity: act_sesh1.activity, user: act_sesh1.user)
       if act_sesh2.blank?
         act_sesh1.update!(classroom_unit_id: cu2.id)
+      elsif act_sesh2.state == ActivitySession::STATE_STARTED && act_sesh1.state == ActivitySession::STATE_FINISHED
```

Comment:
I would move these equality checks into the activity_session.rb file.  One of them is already [written](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/activity_session.rb#L151) and then you could write a similar method for `started?`.  Then the condition becomes `elsif act_sesh2.started? && act_sesh1.finished?`.  This is important since the logic of `started?` is now encapsulated inside of activity_session.rb